File: 백준/Silver/10828. 스택/스택.py
import sys
stack = []
for order in range(int(sys.stdin.readline())):
    order = list(sys.stdin.readline().strip().split())
    # print(order)
    # print(stack)
    if order[0] == 'push':
        # print(order[1])
        stack.append(int(order[1]))
    elif order[0] == 'pop':
        try:
            print(stack.pop(-1))
        except:
            print(-1)
    elif order[0] == 'size':
        print(len(stack))
    elif order[0] == 'empty':
        if len(stack) != 0:
            print(0)
        else:
            print(1)
    elif order[0] == 'top':
        try:
            print(stack[-1])
        except:
            print(-1)

File: node.js/백준/Silver/10828. 스택/스택.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const answer = input.slice(1).reduce(([ans, stack], condition) => {
    const [order, element] = condition.split(' ');
    switch(order) {
        case "push":
            stack.push(Number(element));
            break;
        case "pop":
            ans.push(stack.pop() || -1);
            break;
        case "size":
            ans.push(stack.length);
            break;
        case "empty":
            ans.push(!stack.length ? 1:0);
            break;
        case "top":
            ans.push(stack[stack.length-1] || -1);
            break;
    }
    return [ans, stack];
}, [[], []]);

console.log(answer[0].join('\n'));